aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1192,17 +1170,21 @@ impl AptosVM {
         writeset_payload: &WriteSetPayload,
         txn_sender: Option<AccountAddress>,
         session_id: SessionId,
-    ) -> Result<ChangeSetExt, VMStatus> {
+    ) -> Result<VMChangeSet, VMStatus> {
         let mut gas_meter = UnmeteredGasMeter;
         let change_set_configs =
             ChangeSetConfigs::unlimited_at_gas_feature_version(self.0.get_gas_feature_version());
 
         match writeset_payload {
-            WriteSetPayload::Direct(change_set) => Ok(ChangeSetExt::new(
-                DeltaChangeSet::empty(),
-                change_set.clone(),
-                Arc::new(change_set_configs),
-            )),
+            WriteSetPayload::Direct(change_set) => {
+                let (write_set, events) = change_set.clone().into_inner();
+                VMChangeSet::new(
+                    write_set,
```

Comment:
nit: it may be slightly better to clone write_set and events directly, i.e. change_set.write_set().clone() and change_set.events().clone(), but no strong preference.